main.go

```diff
@@ -13,54 +13,65 @@ import (
 	"github.com/matryer/moq/pkg/moq"
 )
 
+type userFlags struct {
+	outFile string
+	pkgName string
+	args    []string
+}
+
 func main() {
-	var err error
-	defer func() {
```

Comment:
I know `defer` is your favourite keyword in Go but 'Clear is better than clever' and this usage of `defer` I think is quite clever 😄 .